---
title: "Ch10 文件和异常"
category:
 - Programming
tag:
 - Python
---

## 1. 文件

```python
# 10.1.py
# 读取文件

import os

# 获取源文件绝对路径
source_file_path = os.path.realpath(__file__)
# 获取文件夹路径
dir_path = os.path.dirname(source_file_path)
fp = os.path.join(dir_path, "text.txt")

# with 将在文件使用完毕后自动关闭
with open(fp) as file_object:
    contents = file_object.read()  # read 读取全部内容
    print(contents)

# 逐行读取
with open(fp) as fo:
    for line in fo:
        print(line.rstrip())

# 读取到列表中
with open(fp) as fo:
    lines = fo.readlines()

print(lines)

```

```python
# 10.2.py
# 写入文件

from os import path

filename = "write_test.txt"
filepath = path.join(path.dirname(path.realpath(__file__)), filename)

# 写入模式打开文件
with open(filepath, "w") as file_object:
    file_object.write("Hello World!\n")

# 追加模式
with open(filepath, "a") as file_object:
    file_object.write("new content\n")

```

## 2. 异常

使用 `try-except` 处理可能存在的异常

```python
# 10.3.py
# try-expect

a = 1
b = 0
try:
    ans = a / b
except ZeroDivisionError:  # 捕获到异常
    print("cannot divide by zero")
else:  # try 部分正常执行
    print(ans)

```

### 2.1 pass

`pass` 语句表示什么都不做。

```python
try:
    #...
except e:
    pass
```

